test/language/expressions/super/call-proto-not-ctor.js

```diff
@@ -3,43 +3,29 @@
 /*---
 esid: sec-super-keyword
 es6id: 12.3.5
-description: Prototype of active function object must be a constructor
+description: SuperCall should evaluate Arguments prior to checking IsConstructor
 info: |
-  [...]
-  3. Let func be ? GetSuperConstructor().
-
-  12.3.5.2 Runtime Semantics: GetSuperConstructor
+  SuperCall : `super` Arguments
 
   [...]
-  4. Let superConstructor be ? activeFunction.[[GetPrototypeOf]]().
-  5. If IsConstructor(superConstructor) is false, throw a TypeError exception.
+  3. Let _func_ be ! GetSuperConstructor().
+  4. Let _argList_ be ? ArgumentListEvaluation of |Arguments|.
+  5. If IsConstructor(_func_) is *false*, throw a *TypeError* exception.
+  [...]
 features: [class]
 ---*/
 
 var evaluatedArg = false;
-var caught;
 class C extends Object {
   constructor() {
-    try {
-      super(evaluatedArg = true);
-    } catch (err) {
-      caught = err;
-    }
+    super(evaluatedArg = true);
```

Comment:
it seems interesting to preserve the try catch so we can observe the abrupt completion also happening here in the constructor.